src/figuresClasses.ts

```diff
@@ -1,11 +1,94 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  color: string;
+
+  private a: number;
+
+  private b: number;
+
+  private c: number;
+
+  constructor(color: string, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than 0');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error(
+        `The longest side can't be greater than the sum of the other two sides`,
+      );
+    }
+
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: string = 'circle';
 
-export class Triangle implements Figure {}
+  color: string;
 
-export class Circle implements Figure {}
+  private radius: number;
 
-export class Rectangle implements Figure {}
+  constructor(color: string, radius: number) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0');
+    }
+
+    this.color = color;
+    this.radius = radius;
```

Comment:
[CODE KNOWLEDGE] - Similar to the Triangle class, you can simplify the Circle class constructor by using TypeScript's parameter properties feature. This will help you avoid manually repeating the assignments of parameters to class properties.